src/oadp_resources/velero/backup.py

```diff
@@ -1,18 +1,100 @@
+from enum import Enum
+
 from oadp_constants.resources import ApiGroups
 from ocp_resources.resource import NamespacedResource
-from oadp_constants.velero.backup import BackupPhase
-from oadp_utils.wait import wait_for
+from utils.common import wait_for
 
 
 class Backup(NamespacedResource):
     api_group = ApiGroups.VELERO_API_GROUP.value
 
-    def completed(self):
+    # BackupPhase is a str representation of the lifecycle phase
+    # of a Velero backup.
+    class BackupPhase(Enum):
+
+        # BackupPhaseNew means the backup has been created but not
+        # yet processed by the BackupController.
+        New = "New"
+
+        # BackupPhaseFailedValidation means the backup has failed
+        # the controller's validations and therefore will not run.
+        FailedValidation = "FailedValidation"
+
+        # BackupPhaseInProgress means the backup is currently executing.
+        InProgress = "InProgress"
+
+        # BackupPhaseUploading means the backups of Kubernetes resources
+        # and creation of snapshots was successful and snapshot data
+        # is currently uploading.  The backup is not usable yet.
+        Uploading = "Uploading"
+
+        # BackupPhaseUploadingPartialFailure means the backup of Kubernetes
+        # resources and creation of snapshots partially failed (final phase
+        # will be PartiallyFailed) and snapshot data is currently uploading.
+        # The backup is not usable yet.
+        UploadingPartialFailure = "UploadingPartialFailure"
+
+        # BackupPhaseCompleted means the backup has run successfully without
+        # errors.
+        Completed = "Completed"
+
+        # BackupPhasePartiallyFailed means the backup has run to completion
+        # but encountered 1+ errors backing up individual items.
+        PartiallyFailed = "PartiallyFailed"
+
+        # BackupPhaseFailed means the backup ran but encountered an error that
+        # prevented it from completing successfully.
+        Failed = "Failed"
+
+        # BackupPhaseDeleting means the backup and all its associated data are being deleted.
+        Deleting = "Deleting"
+
+    class HookErrorMode(Enum):
+        Continue = "Continue"
+
+        # HookErrorModeFail means that an error from a hook is problematic, and the backup should be in
+        # error.
+        Fail = "Fail"
+
+    def check_phase(self, phase):
```

Comment:
This is a generic function, can you put it in utils (passing the resource object).
A

Reply:
Can be in a different PR

Reply:
ok